### ISSUE TITLE
flowtop sometimes does not show flows after start

### ISSUE THREAD

**Issue by vkochan** [netsniff-ng/netsniff-ng#145]
the issue is happened even if revert to (skip all my flowtop commits):

    git checkout b8a9e53262ab24231542e9ce93daadaf3809dac2

I could reproduce this issue few times, while opening some huge web page with extra
links I start flowtop at the same time and it does not show flows, yes it resets them but after
when I check them with:

    conntrack -L conntrack -p tcp

they are listed there but does not in the flowtop, even if create new connections after.

**Comment by vkochan** [comment]
OK, I did some logging and seems the stucking was caused while flushing IPv6 entries. I make logging:
```
diff --git a/flowtop.c b/flowtop.c
index 0927ff5..d755db3 100644
--- a/flowtop.c
+++ b/flowtop.c
@@ -1121,19 +1121,24 @@ static void *collector(void *null __maybe_unused)
 	struct pollfd poll_fd[1];
 	int ret;
 
+	fprintf(stderr, "Collector start\n");
+
 	ct_event = nfct_open(CONNTRACK, NF_NETLINK_CONNTRACK_NEW |
 				      NF_NETLINK_CONNTRACK_UPDATE |
 				      NF_NETLINK_CONNTRACK_DESTROY);
 	if (!ct_event)
 		panic("Cannot create a nfct handle: %s\n", strerror(errno));
 
+	fprintf(stderr, "Before nfct flush ipv4\n");
 	collector_flush(ct_event, AF_INET);
+	fprintf(stderr, "Before nfct flush ipv6\n");
 	collector_flush(ct_event, AF_INET6);
 
 	filter = nfct_filter_create();
 	if (!filter)
 		panic("Cannot create a nfct filter: %s\n", strerror(errno));
 
+	fprintf(stderr, "Before nfct filter attach\n");
 	ret = nfct_filter_attach(nfct_fd(ct_event), filter);
 	if (ret < 0)
 		panic("Cannot attach filter to handle: %s\n", strerror(errno));
@@ -1161,15 +1166,19 @@ static void *collector(void *null __maybe_unused)
 		nfct_filter_add_attr(filter, NFCT_FILTER_SRC_IPV6, &filter_ipv6);
 	}
 
+	fprintf(stderr, "Before nfct filter attach #2\n");
 	ret = nfct_filter_attach(nfct_fd(ct_event), filter);
 	if (ret < 0)
 		panic("Cannot attach filter to handle: %s\n", strerror(errno));
 
+	fprintf(stderr, "Before nfct filter destroy\n");
 	nfct_filter_destroy(filter);
 
 	nfct_callback_register(ct_event, NFCT_T_ALL, collector_cb, NULL);
 	flow_list_init(&flow_list);
 
+	fprintf(stderr, "Before nfct open\n");
+
 	ct_dump = nfct_open(CONNTRACK, NF_NETLINK_CONNTRACK_UPDATE);
 	if (!ct_dump)
 		panic("Cannot create a nfct handle: %s\n", strerror(errno));
@@ -1187,8 +1196,10 @@ static void *collector(void *null __maybe_unused)
 		panic("Cannot set non-blocking socket: fcntl(): %s\n",
 		      strerror(errno));
 
+	fprintf(stderr, "Before rcu register thread\n");
 	rcu_register_thread();
 
+	fprintf(stderr, "Before while\n");
 	while (!sigint && ret >= 0) {
 		int status;
 
```
and I run flowtop:
$ sudo flowtop/flowtop -s 2> /tmp/flowtop.txt

and when collector is stuck I got the following output from stderr:
[angus@angus-think ~]$ cat /tmp/flowtop.txt 
Collector start
Before nfct flush ipv4
Before nfct flush ipv6

And that explains why my cond wait locking did not work sometimes ...
Will continue investigate why this happens ...

**Comment by vkochan** [comment]
Probably I found the fix but meanwhile I can't really explain why it happens, the fix is to
create separate nfct handle w/o specifying nfnetlink groups ( I assume that something is happened when the same process send FLIUSH event and is a group member who can catch this event) and send FLUSH query via this handle and after close it.

**Comment by tklauser** [comment]
Considering this fixed by 59341681d3dcc3e17105261425b5f08ab954a948. Feel free to reopen if you still see this issue.
